src/gui/downloadfromurldialog.cpp

```diff
@@ -76,19 +77,23 @@ DownloadFromURLDialog::DownloadFromURLDialog(QWidget *parent)
     const QString clipboardText = qApp->clipboard()->text();
     const QList<QStringView> clipboardList = QStringView(clipboardText).split(u'\n');
 
-    QSet<QString> uniqueURLs;
-    for (QStringView strRef : clipboardList)
+    // show urls in the original order as from the clipboard
+    QStringList urls;
+    urls.reserve(clipboardList.size());
+
+    for (QStringView url : clipboardList)
```

Comment:
Isn't all this preprocessing pointless? Firstly, it really confuses the user, since it is seen something different from what was copied to the clipboard. Secondly, the user can still change the content directly in the dialog, thereby making invalid elements again. So maybe it makes sense to postpone processing to the final stage? Then the automatically inserted content will look exactly the same as if the user personally inserted the content from the clipboard (Ctrl+V).

From the other side this can be considered as "Auto detecting suitable URLs in Clipboard" so preprocessed content is OK. But in this case, I would expect to deal not with a plain text editor, but with more advanced list-based control that will prevent user from insertion invalid/duplicate items.

Reply:
>From the other side this can be considered as "Auto detecting suitable URLs in Clipboard" so preprocessed content is OK.

Yes, this is the intention here.

>But in this case, I would expect to deal not with a plain text editor, but with more advanced list-based control that will prevent user from insertion invalid/duplicate items.

This PR now allows duplicates when fetching from clipboard and the duplicates will be filtered out (behind the scene) when user submits the dialog. This behavior is better than the previous one.

>Firstly, it really confuses the user, since it is seen something different from what was copied to the clipboard.

I thought about it too. However it came to me that this isn't just copy the clipboard (which is unexpected behavior IMO) but some kind of smart detection.